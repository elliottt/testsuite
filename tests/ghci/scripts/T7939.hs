{-# LANGUAGE TypeFamilies, PolyKinds #-}

module T7939 where

class Foo a where
   type Bar a


