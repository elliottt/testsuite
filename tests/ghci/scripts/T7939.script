:l T7939
:i Bar
:k Bar
