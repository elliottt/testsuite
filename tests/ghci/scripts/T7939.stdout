class Foo (k :: BOX) (a :: k) where
  type family Bar (k :: BOX) (a :: k) b :: *
  	-- Defined at T7939.hs:6:9
Bar :: k -> * -> *
type family F a :: * 	-- Defined at T7939.hs:8:13
type instance F Int -- Defined at T7939.hs:9:1
F :: * -> *
type family G a :: * where G Int = Bool
  	-- Defined at T7939.hs:11:13
G :: * -> *
type family H (a :: Bool) :: Bool where H 'False = 'True
  	-- Defined at T7939.hs:14:13
H :: Bool -> Bool
type family J (k :: BOX) (a :: [k]) :: Bool where
    J k ('[] k) = 'False
    J k ((':) k h t) = 'True
  	-- Defined at T7939.hs:17:13
J :: [k] -> Bool
type family K (k :: BOX) (a :: [k]) :: Maybe k where
    K k ('[] k) = 'Nothing k
    K k ((':) k h t) = 'Just k h
  	-- Defined at T7939.hs:21:13
K :: [k] -> Maybe k
